src/colorizer/shaders/colorize_RGBA8U.frag

```diff
@@ -87,25 +156,55 @@ vec4 alphaBlend(vec4 a, vec4 b) {
   return vec4((a.rgb * a.a + b.rgb * b.a * (1.0 - a.a)) / alpha, alpha);
 }
 
-vec4 getMainPixelColor() {
+bool isOutsideBounds(vec2 sUv) {
+  return sUv.x < 0.0 || sUv.y < 0.0 || sUv.x > 1.0 || sUv.y > 1.0;
+}
+
+vec4 getBackdropColor(vec2 sUv) {
+  if (isOutsideBounds(sUv)) {
+    return TRANSPARENT;
+  }
+  vec4 backdropColor = texture(backdrop, sUv).rgba;
+  // Switch between HSV and HSL color adjustments based on whether we are increasing or
+  // decreasing brightness. HSL is better for increasing brightness, since it does not increase
+  // perceptual saturation when lightness increases (the way HSV does when value increases).
+  // HSV is better than HSL for decreasing brightness for similar reasons.
```

Comment:
Interesting. I'm surprised that HSV is not good enough

Reply:
This may be a symptom of me using value/lightness as a proxy for brightness and not looking for an actual algorithmic approach 😅

![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/06ce5bf3-9872-435e-bdad-c839fc0b1cc4)

If we're increasing lightness/value, then HSV causes colors to become unexpectedly more saturated.

![image](https://github.com/allen-cell-animated/nucmorph-colorizer/assets/30200665/a5baebc2-4e1f-46d8-b41f-619b56ef0c83)

When decreasing lightness/value, HSL can sometimes cause colors to become more saturated, which is most evident for very light, off-white colors.

Reply:
I did also find this algorithm, which is what GIMP uses: https://en.wikipedia.org/wiki/Image_editing#Contrast_change_and_brightening

Reply:
that algorithm looks so much simpler, I kinda like that. just per-rgb without changing color space.  if it works for gimp.... ?  From the pictures you show, it seems clear that L and V are not quite right and I can see why you do the conditional.  From those pictures, L looks much more appropriate than V as a brightness control.  But since the gimp code is so simple, you could try it and see if it feels right.

Reply:
It works!